ExampleModels/Arithmetic/src/arithmetic.jl

```
@@ -0,0 +1,107 @@
+module Arithmetic
+
+abstract type Expression end
+
+struct Constant <: Expression
+    value::Number
+end
+
+@enum OP begin
+    ADD = 1
+    SUB = 2
+    MUL = 3
+    DIV = 4
+end
+
+struct BinaryOp <: Expression
+    left::Expression
+    op::OP
+    right::Expression
+end
+
+function evaluate(expr::Constant)
+    expr.value
+end
+
+function evaluate(expr::BinaryOp) 
+    leftVal = evaluate(expr.left)
+    rightVal = evaluate(expr.right)
+    if expr.op == ADD
+        return leftVal + rightVal
+    elseif expr.op == SUB
+        return leftVal - rightVal
+    elseif expr.op == MUL
+        return leftVal * rightVal
+    elseif expr.op == DIV
+        return leftVal / rightVal
+    else 
+        println("Unknown op while evaluating BinaryOp")
```

throw an exception, instead of `println`